tests/stdlib/topenssl.nim

```diff
@@ -0,0 +1,19 @@
+discard """
+  matrix: "-d:ssl"
+"""
+import openssl
+
+# bug #16308
+block:
+  var rsa: PRSA
+  var file: File
+
+  if not file.open("topenssl_rsa_public_key.pem"):

```

Comment:
bug: how can this even work? tests are run from nim root, not currentSourcePath.parentDir. But you don't even have to assume this; instead use: `let path = currentSourcePath.parentDir / "topenssl_rsa_public_key.pem"` which is robust